checkov/policies_3d/checks_parser.py

```diff
@@ -18,3 +37,121 @@ def parse_raw_check(self, raw_check: Dict[str, Dict[str, Any]], **kwargs: Any) -
+        cve_definition = [definition["cves"] for definition in self.check_definition if "cves" in definition.keys()][0]
+        if not cve_definition:
+            return None
+
+        if not any(op in cve_definition.keys() for op in SUPPORTED_LOGICAL_OPERATORS):
+            return None
+
+        top_level_predicament = Predicament(logical_op='')
+        top_level_logical_op = ''
+        for key, value in cve_definition.items():
+            if key in SUPPORTED_LOGICAL_OPERATORS:
+                top_level_logical_op = key
+                top_level_predicament.logical_op = key
+            elif key == 'risk_factors':
+                value = [value] if isinstance(value, str) else value
+                top_level_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
+
+        nested_definition = cve_definition[top_level_logical_op]
+        nested_logical_op = None
+        for key, value in nested_definition[0].items():
+            if key in SUPPORTED_LOGICAL_OPERATORS:
+                nested_logical_op = key
+            elif key == 'risk_factors':
+                value = [value] if isinstance(value, str) else value
+                top_level_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
+
+        nested_predicament = None
+        if nested_logical_op:
+            nested_definition = nested_definition[0][nested_logical_op]
+            if not isinstance(nested_definition, list):
+                return None
+
+            nested_predicament = Predicament(logical_op=nested_logical_op)
+            for definition in nested_definition:
+                for key, value in definition.items():
+                    if key == 'risk_factors':
+                        value = [value] if isinstance(value, str) else value
+                        nested_predicament.predicates.append(RiskFactorCVEContains(value, cve_report))
+
+        if nested_predicament:
+            top_level_predicament.predicaments.append(nested_predicament)
+
+        return top_level_predicament
+
+
+
+    def _create_iac_predicament(self, iac_record: Record) -> Predicament | None:
+        iac_definition = [definition["iac"] for definition in self.check_definition if "iac" in definition.keys()][0]
+        if not iac_definition:
+            return None
+
+        if not any(op in iac_definition.keys() for op in SUPPORTED_LOGICAL_OPERATORS):
+            return None
+
+        top_level_predicament = Predicament(logical_op='')
+        top_level_logical_op = ''
+        for key, value in iac_definition.items():
+            if key in SUPPORTED_LOGICAL_OPERATORS:
+                top_level_logical_op = key
+                top_level_predicament.logical_op = key
+            elif key == 'violation_id':
+                value = [value] if isinstance(value, str) else value
+                top_level_predicament.predicates.append(ViolationIdEquals(iac_record, value))
```

Comment:
The extracted function would also help here. So it could also get a value of "type" and then it would get the mapping by itself of cve / secret / iac depending on the type. The rest of the logic is identical